server.js

```diff
@@ -79,7 +82,45 @@ app.get(/^\/([^.]+)$/, function (req, res) {
 
 // start the app
 
-app.listen(port);
-console.log('');
-console.log('Listening on port ' + port);
 console.log('');
+// Check that default port is free, else offer to change
+portScanner.findAPortNotInUse(port, port+50, '127.0.0.1', function(error, availablePort) {
+
+  if (port == availablePort){
+    app.listen(port);
+    console.log('Listening on port ' + port + '   url: http://localhost:' + port);
+  }
+  else {
+    // Default port in use - offer to change to available port
+    console.log("ERROR: Port " + port + " in use - do you have another prototype running?\n");
+    // Set up prompt settings
+    prompt.colors = false;
+    prompt.start();
+    prompt.message = "";
+    prompt.delimiter = "";
+
+    // Ask user if they want to change port
+    prompt.get([{
+      name: 'answer',
+      description: 'Change to an available port? (y/n)',
+      required: true,
+      type: 'string',
+      pattern: /y(es)?|no?/i,
+      message: 'Please enter y or n'
+    }], function (err, result) {
+      if (result.answer.match(/y(es)?/i) ) {
+        // User answers yes
+        port = availablePort;
+        app.listen(port);
+        console.log('Changed to port ' + port + '   url: http://localhost:' + port);
+      }
+      else {
+        // User answers no - exit
+        console.log('You can set a new default port in server.js, or by running the server with PORT=XXXX');
+        console.log("Exiting");
+        process.exit(0);
```

Comment:
That's no worse than what it did before (though previously it exited with code `1`. I wasn't sure how to exit nodemon - if you know, can you add it?